release/ray_release/test_automation/state_machine.py

```diff
@@ -50,10 +72,48 @@ def _move_hook(self, from_state: TestState, to_state: TestState) -> None:
         Action performed when test transitions to a different state. This is where we do
         things like creating and closing github issues, trigger bisects, etc.
         """
-        if from_state == TestState.PASSING and to_state == TestState.FAILING:
+        if (
+            from_state == TestState.PASSING
+            and to_state == TestState.CONSITENTLY_FAILING
+        ):
+            self._create_github_issue()
+        elif (
+            from_state == TestState.FAILING
+            and to_state == TestState.CONSITENTLY_FAILING
+        ):
             self._create_github_issue()
-        elif from_state == TestState.FAILING and to_state == TestState.PASSING:
+        elif (
+            from_state == TestState.CONSITENTLY_FAILING
+            and to_state == TestState.PASSING
+        ):
             self._close_github_issue()
+        elif from_state == TestState.PASSING and to_state == TestState.FAILING:
+            self._trigger_bisect()
+
+    def _trigger_bisect(self) -> None:
+        build = self.ray_buildkite.builds().create_build(
+            "ray-project",
+            "release-tests-bisect",
+            "HEAD",
+            "master",
+            message=f"Triggered by test {self.test.get_name()} failing",
+        )
+        failing_commit = self.test_results[0].commit
+        passing_commit = [r.commit for r in self.test_results if r.is_passing()][0]
+        self.ray_buildkite.jobs().unblock_job(
+            "ray-project",
+            "release-tests-bisect",
+            build["number"],
+            build["jobs"][0]["id"],  # first job is the blocked job
+            fields={
+                "test-name": self.test.get_name(),
+                "passing-commit": passing_commit,
+                "failing-commit": failing_commit,
```

Comment:
do we worry that this might spin up too many tests? do we want to add a threshold of some sort to limit the number of tests that gets auto scheduled (if this is auto scheduling)?

Reply:
i think we should, we need a rate limit here